src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/SimplifyBooleanExpressionCheck.java

```diff
@@ -105,17 +128,39 @@
     @Override
     public void visitToken(DetailAST ast) {
         final DetailAST parent = ast.getParent();
-        switch (parent.getType()) {
-            case TokenTypes.NOT_EQUAL:
-            case TokenTypes.EQUAL:
-            case TokenTypes.LNOT:
-            case TokenTypes.LOR:
-            case TokenTypes.LAND:
-                log(parent, MSG_KEY);
-                break;
-            default:
-                break;
+        if (TokenUtil.isOfType(parent, UNACCEPTABLE_PARENT_OF_TRUE_OR_FALSE)
+                || parent.getType() == TokenTypes.QUESTION
+                && canExprWithTernaryOperatorBeSimplified(ast, parent)) {
+            log(parent, MSG_KEY);
         }
     }
 
+    /**
+     * Whether the expression with ternary operator can be simplified.
+     *
+     * @param ast ast
+     * @param parentAst parentAst
+     * @return {@code true} if expression with ternary operator can be simplified
+     */
+    private static boolean canExprWithTernaryOperatorBeSimplified(
+            DetailAST ast, DetailAST parentAst) {
+        final DetailAST nextSibling = ast.getNextSibling();
+
+        return TokenUtil.isOfType(parentAst.getFirstChild(), TRUE_AND_FALSE_TOKEN)
+                || isSecResultantExprLiteralTrueOrFalse(nextSibling);
```

Comment:
`isSecResultantExprLiteralTrueOrFalse` -> `isBooleanLiteral`

Reply:
In general, you can extract check for literals and make code look like
```
return isBooleanLiteral(parentAst.getFirstChild()) || nextSibling != null && isBooleanLiteral(nextSibling.getNextSibling();
```

Reply:
done